src/components/StepsMonitor.tsx

```diff
@@ -181,39 +125,38 @@ function CodeFlyout({
   );
 }
 
-export function StepsMonitor() {
+interface IStepsMonitor {
+  isFlyoutVisible: boolean;
+  setIsFlyoutVisible: Setter<boolean>;
+}
+
+export function StepsMonitor({
+  isFlyoutVisible,
+  setIsFlyoutVisible,
+}: IStepsMonitor) {
   const { actions } = useContext(StepsContext);
-  const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
   const [code, setCode] = useState("");
   const [type, setType] = useState<JourneyType>("inline");
 
   return (
-    <>
-      <EuiPanel
-        color="transparent"
-        borderRadius="none"
-        style={{ minHeight: 500 }}
-      >
-        <Steps />
-        <EuiSpacer />
-
-        {isFlyoutVisible && (
-          <CodeFlyout
-            actions={actions}
-            code={code}
-            setCode={setCode}
-            setIsFlyoutVisible={setIsFlyoutVisible}
-            setType={setType}
-            type={type}
-          />
-        )}
-      </EuiPanel>
-      <StepsFooter
-        actions={actions}
-        setCode={setCode}
-        setIsFlyoutVisible={setIsFlyoutVisible}
-        type={type}
-      />
-    </>
+    <EuiPanel
+      color="transparent"
+      borderRadius="none"
+      style={{ minHeight: 500 }}
+    >
+      <Steps />
+      <EuiSpacer />
+
+      {isFlyoutVisible && (
+        <CodeFlyout
+          actions={actions}
+          code={code}
+          setCode={setCode}
+          setIsFlyoutVisible={setIsFlyoutVisible}
+          setType={setType}
+          type={type}
```

Comment:
[Nit] Should we manage the type selection within the `CodeFlyout` itself given it's not used on the this component?

Reply:
We can maintain this state wholly in the flyout now. Previously it was shared across the app but I think now it's only in the code flyout.

Reply:
Done in https://github.com/elastic/synthetics-recorder/pull/107/commits/2b550099ffb26fdc1f312940e502c456eccd8f68.